src/core/QuoridorGame.cs

```diff
@@ -1,16 +1,55 @@
 ﻿using QuoridorEngine.Solver;
+using QuoridorEngine.Utils;
+using System.Diagnostics;
 
 namespace QuoridorEngine.Core
 {
-    internal class QuoridorGame : IGameState
+    /// <summary>
+    /// Implementation of the IGameState interface specific to
+    /// a Quoridor state. Is responsible for managing all the 
+    /// information and operations related to a Quoridor Game
+    /// including but not limited to the board, the players,
+    /// the walls, the moves etc. It's only possible to interact
+    /// with a Quoridor Game State using this class' public interface
+    /// </summary>
+    public class QuoridorGame : IGameState
     {
+        private readonly int dimension = 9;
+        private readonly QuoridorBoard board;
+        private readonly QuoridorPlayer white;
+        private readonly QuoridorPlayer black;
+        private readonly List<QuoridorMove> gameHistory;
+
+        /// <summary>
+        /// Initializes a new Quoridor Game with the specified parameters
+        /// </summary>
+        /// <param name="dimension">The dimension of the board.
+        /// Needs to be odd an greater than 2</param>
+        public QuoridorGame(int dimension)
+        {
+            Debug.Assert(dimension > 2);
+            Debug.Assert(dimension % 2 == 1);
```

Comment:
Since players interact directly with the QuoridorGame class maybe we should implement an InvalidBoardDimentions exception instead of using assertions